### PR TITLE
feat: introduce sample batches for krakenuniq

### PR DESCRIPTION
**Pull request by Midnighter**

### CODE REVIEW

**Comment by Midnighter** [comment]
I probably forgot to document the new parameter somewhere but in principle this seems to be working.

**Review by jfy133 — COMMENTED** [review]
- Missing CHANGELOG
- I wonder if we should add to a FAQ section of Usage describing the motivation and guidance how/when to use it?
    - i.e. it still makes me nervous as I would fear two batches going to the same node and possibly having a database conflict as there are two in memory at once [although this is assumed, maybe it's not an issue and there is some magic going on]

I'm going to try and test it now.

Another question, what happens if you supply a number for the batch size that is more than the number of files (if you just want to make sure you always do a single batc?h?)

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### `Added`
 
+- [#276](https://github.com/nf-core/taxprofiler/pull/276) Implemented batching in the KrakenUniq samples processing.
```
Comment (jfy133) [inline CHANGELOG.md:10]:
```suggestion
- [#276](https://github.com/nf-core/taxprofiler/pull/276) Implemented batching in the KrakenUniq samples processing. (added by @Midnighter)
```

File: nextflow_schema.json [hunk 2]
```diff
@@ -426,6 +426,13 @@
                     "description": "Turn on saving of KrakenUniq per-read taxonomic assignment file",
                     "help_text": "Save a text file that contains a list of each read that had a taxonomic assignment, with information on specific taxonomic taxonomic assignment that that read recieved.\n\n> Modifies tool parameter(s):\n> - krakenuniq: `--output`"
                 },
+                "krakenuniq_batch_size": {
+                    "type": "integer",
+                    "default": 20,
+                    "fa_icon": "far fa-window-restore",
+                    "description": "Set the batch size for KrakenUniq",
```
Comment (jfy133) [inline nextflow_schema.json:433]:
```suggestion
                    "description": "Specify the number samples for each KrakenUniq run",
```

File: nextflow_schema.json [hunk 3]
```diff
@@ -426,6 +426,13 @@
...
+                    "help_text": "Specify the batch size for KrakenUniq. The reference database for KrakenUniq is loaded into memory once in a process and then used to classify many samples. This parameter determines for how many samples at a time."
```
Comment (jfy133) [inline nextflow_schema.json:434]:
```suggestion
                    "help_text": "Specify the batch size for KrakenUniq. The reference database for KrakenUniq is loaded into memory once in a process and then used to classify many samples. When you have many samples, a single KrakenUniq run can be rather slow. Alternatively, we can split up KrakenUniq runs for a 'batch' of samples, allowing a balance between shared using a database for multiple samples, but also faster parallelised KrakenUniq runs. This parameter determines for how many samples at a time."
```

To give more context why you would want to set this.

**Review by jfy133 — APPROVED** [review]
Some docs tweaks, but otherwise we have an OK that it's working (as reported on slack) it's good to go now I think.

**Comment by Midnighter** [comment]
> @Midnighter do you know why the tests aren't running thinking

No, but they'll run again when I add the changes.

**Pull request merged**